### PR TITLE
'Send' link on keep card

### PR DESCRIPTION
**Pull request by martinraison**
@2is10 @andrewconner @stkem

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular-black/src/keep/keep.tpl.html [hunk 1]
```diff
@@ -37,13 +37,11 @@
...
   </div>
   <div class="kf-keep-footer">
-    <span class="kf-keep-footer-link"><a>Send</a></span>
-    <span class="kf-keep-footer-link" ng-if="!hasTag() && isMyBookmark(keep)" ng-click="showAddTag()"><a>Tag</a></span>
-    <span class="kf-keep-footer-link" ng-click="togglePrivate()" ng-if="isMine() && !isPrivate()"><a>Make Private</a></span>
-    <span class="kf-keep-footer-link" ng-click="togglePrivate()" ng-if="isMine() && isPrivate()"><a>Make Public</a></span>
-    <span class="kf-keep-footer-link" ng-if="isMine()" ng-click="unkeep()"><a>Unkeep</a></span>
-    <span class="kf-keep-footer-link" ng-if="!isMine()" ng-click="keepPublic()"><a>Keep</a></span>
-    <span class="kf-keep-footer-link" ng-if="!isMine()" ng-click="keepPrivate()"><a>Private Keep</a></span>
+    <span class="kf-keep-footer-link"><a ng-href="{{keep.url}}" data-locator='/messages#compose'>Send</a></span>
```
Comment (martinraison) [inline kifi-backend/modules/shoebox/angular-black/src/keep/keep.tpl.html]: is the locator correct? (it works for me)

Comment (2is10) [inline kifi-backend/modules/shoebox/angular-black/src/keep/keep.tpl.html]: Yes, it’s correct.
`/messages:all#compose` would take them to their all-messages pane underneath the compose.
Just `#compose` might also do the same. I aimed to make the extension flexible.

Comment (martinraison) [inline kifi-backend/modules/shoebox/angular-black/src/keep/keep.tpl.html]: I see, nice - thanks!

**Comment by stkem** [comment]
Can we detect wether the extension is installed and show a message if it isn't if someone clicks this? If it't not installed, this will just open the page and do nothing else, which might be a bit confusing the the user.

**Comment by martinraison** [comment]
I guess we can - let me have a look

**Comment by martinraison** [comment]
Oh yes - we have the install service already

**Comment by 2is10** [comment]
You’ll want to make sure `data-kifi-ext="3.0.7"` in the `<html>` element is 3.0.7 or higher to show the Send links. (I assume the install service parses and exposes the version.)

P.S. Can’t wait till we allow off-page messaging. I think a better UI for this would be to open a compose form inline, with the option to pop it over to the actual page (e.g. for a look here). I don’t think look heres will be common in messages sent using these “Send” links.

**Pull request merged**